### PR TITLE
Updating diktat to 1.2.1 and it's the configuration

### PR DESCRIPTION
**Pull request by akuleshov7**
And same for detekt

### CODE REVIEW

**Comment by akuleshov7** [comment]
@CommanderTvis answering your question

File: reflekt-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/generation/ir/SmartReflektIrTransformer.kt [hunk 1]
```diff
@@ -42,7 +42,7 @@ class SmartReflektIrTransformer(
     private val messageCollector: MessageCollector? = null,
 ) : BaseReflektIrTransformer(messageCollector) {
     private val importChecker = ImportChecker(classpath)
-    private val sources = HashMap<String, SourceFile>()
+    private val sources: HashMap<String, SourceFile> = HashMap()
```
Comment (CommanderTvis) [inline reflekt-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/generation/ir/SmartReflektIrTransformer.kt:45]: Usually, when the type of a collection property is specified explicitly, a more abstract type is used:
```suggestion
    private val sources: MutableMap<String, SourceFile> = HashMap()
```

File: .github/workflows/diktat.yml [hunk 2]
```diff
@@ -10,14 +10,37 @@ jobs:
...
-      - uses: gradle/gradle-build-action@v2.1.5
+          distribution: zulu
+      - uses: gradle/gradle-build-action@v2
+        with:
+          gradle-version: wrapper
+          arguments: |
+            diktatCheck
+            -Pdiktat.githubActions=true
+            -Pdetekt.multiplatform.disabled=true
+      # Upload reports only if code style checks were failed
```
Comment (CommanderTvis) [inline .github/workflows/diktat.yml:28]: I think it is OK to upload a SARIF report even when it's without errors.

**Review by CommanderTvis — CHANGES_REQUESTED** [review]
Also, merge conflicts have to be resolved.

**Comment by nbirillo** [comment]
@akuleshov7 could you please resolve the conflicts?

File: .github/workflows/diktat.yml (hunk 2 above)
Comment (akuleshov7) [reply .github/workflows/diktat.yml:28]: ok, np

**Pull request closed without merging**